contracts/contracts/harvest/OETHBaseHarvester.sol

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Governable } from "../governance/Governable.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IVault } from "../interfaces/IVault.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { ISwapRouter } from "../interfaces/aerodrome/ISwapRouter.sol";
+
+contract OETHBaseHarvester is Governable {
+    using SafeERC20 for IERC20;
+
+    IVault public immutable vault;
+    IStrategy public immutable amoStrategy;
+    IERC20 public immutable aero;
+    IERC20 public immutable weth;
+    ISwapRouter public immutable swapRouter;
+
+    // Similar sig to `AbstractHarvester.RewardTokenSwapped` for
+    // future compatibility with monitoring
+    event RewardTokenSwapped(
+        address indexed rewardToken,
+        address indexed swappedInto,
+        uint8 swapPlatform,
+        uint256 amountIn,
+        uint256 amountOut
+    );
+
+    /**
+     * @notice Verifies that the caller is either Governor or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            msg.sender == vault.strategistAddr() || isGovernor(),
+            "Caller is not the Strategist or Governor"
+        );
+        _;
+    }
+
+    constructor(
+        address _vault,
+        address _amoStrategy,
+        address _aero,
+        address _weth,
+        address _swapRouter
+    ) {
+        vault = IVault(_vault);
+        amoStrategy = IStrategy(_amoStrategy);
+        aero = IERC20(_aero);
+        weth = IERC20(_weth);
+        swapRouter = ISwapRouter(_swapRouter);
+    }
+
+    /**
+     * @notice Collects AERO from AMO strategy and
+     *      sends it to the Strategist multisig.
+     *      Anyone can call it.
+     */
+    function harvest() external {
```

Comment:
Minor thing - anyone can call harvest.

If we are trying to operate by using harvestAndSwap(), harvest() being before called would cancel the swap or remove most of the areo from being swappable. Over any given time range we'd probably be operating either in auto harvest() mode, or in manual harvestAndSwap() mode, but overlapping both calls.

It's probably not worth adding any complications around locking out harvest though.

Reply:
I would argue that it is worth adding a "harvestEnabled" boolean settable by the strategist. This way we can disable the public callable harvest function when required. I know it is additional complexity - which I also don't like.

But just 1 malicious actor is required and they can disable the convenient harvestAndSwap function for us (by calling harvest()) and we need to perform swapping and distributing the fees manually. Granted there is little incentive for anyone to be doing such a thing.

Reply:
That's a valid concern. However I don't think calling `harvest()` before we do a `harvestAndSwap` disables us from using that function. We can always transfer AERO to that contract in the same tx as `harvestAndSwap` to use it.

I'm trying to avoid any sort of storage variables to not mess up the storage layout when we start inheriting `AbstractHarvester` in future. One alternative I can think of is:
- in `harvestAndSwap`, when there isn't enough funds in the contracts, it can do a `transferFrom(strategist, address(this))` for the difference. This means that strategist has to approve the Harvester contract to move AERO on its behalf but that should be fine I guess. 

What do you think?

Reply:
I have went ahead made the change to `transferFrom`, let me know if you have any further comments/suggestion on this one

Reply:
I just made the `harvest()` method permissioned in addition to the above change. This concern should no longer be there